src/include/gpopt/translate/CTranslatorRelcacheToDXL.h

```diff
@@ -18,16 +18,18 @@
 #ifndef GPDXL_CTranslatorRelcacheToDXL_H
 #define GPDXL_CTranslatorRelcacheToDXL_H
 
-#include "gpos/base.h"
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wpedantic"
 
 extern "C" {
 #include "postgres.h"
-
-#include "access/tupdesc.h"
-#include "catalog/gp_distribution_policy.h"
-#include "foreign/foreign.h"
 }
 
+#pragma GCC diagnostic pop
```

Comment:
postgres.h is included from gpdbdefs.h. Can we delete these lines?

Reply:
```
Note 2: 'CTranslatorRelcacheToDXL.h' still has a separate include of
'postgres.h' at the beginning, as the consequent includes rely on the defines
from this file. Without it in the first place, bfv_statistic test will fail. And
we can't move 'gpdbdefs.h' to the file start instead of it due to the
'clang-format' check.
```
Before the change gpos/base.h was included before postgres.h. How did it work?

Reply:
Because postgres.h was included in the [CTranslatorRelcacheToDXL.cpp](https://github.com/arenadata/gpdb/pull/1106/files#diff-8af69ff652dc86195110245bc462a2569384868c743aff9dda83357f6ff578f6) at a first place

Reply:
I suggest to include postgres.h to gpos/base.h, if postgres.h is necessary for it, because include directives order should not matter.